src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -3415,6 +3415,19 @@ int SketchObject::addSymmetric(const std::vector<int> &geoIdList, int refGeoId,
     std::map<int, int> geoIdMap;
     std::map<int, bool> isStartEndInverted;
 
+    // Find out if reference is aligned with V or H axis,
+    // if so we can keep Vertical and Horizontal constrants in the mirrored geometry.
+    bool refIsAxisAligned = false;
+    if (refGeoId == Sketcher::GeoEnum::VAxis || refGeoId == Sketcher::GeoEnum::HAxis)
+        refIsAxisAligned = true;
+    else {
+        for (std::vector<Constraint *>::const_iterator it = constrvals.begin(); it != constrvals.end(); ++it) {
+            Constraint *constr = *(it);
+            if (constr->First != refGeoId && (constr->Type == Sketcher::Vertical || constr->Type == Sketcher::Horizontal))

```

Comment:
Not sure this is the intended code, don't you want:
```
if (constr->First == refGeoId && (constr->Type == Sketcher::Vertical || constr->Type == Sketcher::Horizontal))
```
I mean, before you had an "if" that would continue if constr->First != refGeoId, so the second if would only evaluate if constr->First == refGeoId. It also appears that you want to set the refIsAxisAligned = true if refGeoID is constr->First.

Reply:
Gosh, the good old "this change is so simple that I don't have to think about it nor test it". Will fix.

Reply:
Ok, fixed.